src/asm2wasm.h

```diff
@@ -1374,16 +1382,13 @@ void Asm2WasmBuilder::processAsm(Ref ast) {
 
   ModuleUtils::iterImportedFunctions(wasm, [&](Function* import) {
     IString name = import->name;
-    if (importedFunctionTypes.find(name) != importedFunctionTypes.end()) {
+    if (importedSignatures.find(name) != importedSignatures.end()) {
       // special math builtins
-      FunctionType* builtin =
-        getBuiltinFunctionType(import->module, import->base);
-      if (builtin) {
-        import->type = builtin->name;
+      Signature builtin = getBuiltinSignature(import->module, import->base);
+      if (builtin != Signature()) {
```

Comment:
Yeah, this is rather ugly. Empty signature is indeed (none, none). I thought that was what the default constructor was doing, but I guess I don't know for sure. I'll change it to explicitly initialize params and results to none.

`getBuiltinSignature` is meant to return an optional signature, and in the absence of `std::optional` I was just using the default signature to mean the absence of a builtin signature. But this is an ugly hack because there presumably could at some point be a builtin JS function with signature (none, none). I will change this to take a pointer to the result signature and return a bool indicating whether the signature exists. This is a more idiomatic ugly hack.

Reply:
Hmm, I see. About the default constructor of `Type`, wouldn't appending `= 0;` to [this line](https://github.com/WebAssembly/binaryen/blob/247884e68b30dedeb01457d6c7e50d4bba5e4abc/src/wasm-type.h#L27) be sufficient?